### PR TITLE
Checking for already offered or already chosen driver

### PR DESCRIPTION
**Pull request by Hreherch**
I haven't tested yet, but I think this is a fix. (Squash if you test and agree)

### CODE REVIEW

**Comment by Hreherch** [comment]
Still need to check if the checks are working for offering, but I'm suddenly a lot more relieved.

**Comment by Hreherch** [comment]
Ready for merge/review
Solves: #148, #134, #131, #151 

@sehoven, @MandyMeindersma, @kieter, @MichaelParadis

**Comment by Hreherch** [comment]
Pull request now also implements code for #127 from merged branch #152.

File: app/src/androidTest/java/comcmput301f16t01/github/carrier/StatusTests.java [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import comcmput301f16t01.github.carrier.Requests.ElasticRequestController;
 import comcmput301f16t01.github.carrier.Requests.Request;
 import comcmput301f16t01.github.carrier.Requests.RequestController;
+import comcmput301f16t01.github.carrier.Users.User;
```
Comment (MandyMeindersma) [inline app/src/androidTest/java/comcmput301f16t01/github/carrier/StatusTests.java:8]: yay beautiful imports

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 2]
```diff
@@ -14,29 +14,35 @@
 import android.support.v4.app.FragmentPagerAdapter;
 import android.support.v4.content.ContextCompat;
 import android.support.v4.view.ViewPager;
+import android.support.v4.widget.SwipeRefreshLayout;
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.util.Log;
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:22]: Do we still need logs now? its just debugging?

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 3]
```diff
@@ -284,7 +281,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
      * Pop up a AlertDialog to confirm and open a new LoginActivity, while closing the current
      * RiderMainActivity.
      */
-    public void onBackPressed() {
+    public void logout() {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:295]: update the issue <3

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 4]
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
             if( getArguments().getInt(ARG_SECTION_NUMBER) == 1 ) {
                 fillRiderRequests( requestListView );
             } else {
                 fillDriverRequests( requestListView );
             }
 
+            final SwipeRefreshLayout srl = (SwipeRefreshLayout) rootView.findViewById( R.id.swiperefresh );
+
+            /* src: http://stackoverflow.com/questions/26295481/android-swiperefreshlayout-how-to-implement-canchildscrollup-if-child-is-not-a-l
+             * user: iamlukeyb
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:379]: is this in the wiki?

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 5]
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                }
+            });
+
+            // Set up SwipeRefresh and what should happen on a swipe action
+            srl.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+                @Override
+                public void onRefresh() {
+                    Log.i( "onCreateView", "onRefresh called from SwipeRefreshLayout");
+
+                    if(!ConnectionChecker.isConnected(getContext())) {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:400]: cool. Good offline stuff

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 6]
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                        Toast.makeText(getContext(), "You have no network connection!", Toast.LENGTH_LONG ).show();
+                        srl.setRefreshing( false );
+                        return;
+                    }
+
+                    // Checks for when the AsyncTask is finished
+                    ElasticRequestController.setListener(new Listener() {
+                        private int finish = 0;
+                        @Override
+                        public void update() {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:412]: What is finish? and why is it 2? can we comment here?

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 7]
```diff
@@ -355,38 +352,88 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+
             requestListView.setAdapter(requestArrayAdapter);
 
-            // add listener to update this view
-            rc.addListener( new Listener() {
+            requestList.addListener(new Listener() {
                 @Override
                 public void update() {
                     requestArrayAdapter.notifyDataSetChanged();
+                    //mSectionsPagerAdapter.notifyDataSetChanged();
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java]: delete this line now?

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 8]
```diff
@@ -399,7 +446,8 @@ public void update() {
                 public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                     Intent intent = new Intent(getActivity(), DriverViewRequestActivity.class);
                     Bundle bundle = new Bundle();
-                    bundle.putString("request", new Gson().toJson(requestList.get(position)));
+                    //bundle.putString("request", new Gson().toJson(requestList.get(position)));
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java]: delete this now?

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 9]
```diff
@@ -411,27 +459,24 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
...
-            // add a listener to listen to changes and update this view.
-            rc.addListener(new Listener() {
+            requestList.addListener(new Listener() {
                 @Override
                 public void update() {
                     requestArrayAdapter.notifyDataSetChanged();
                 }
             });
-    
+
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java:492]: I AM SARAH DO NOT LEAVE BLANK LINES

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 10]
```diff
@@ -441,8 +486,8 @@ public void update() {
                 public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                     Intent intent = new Intent(getActivity(), RiderRequestActivity.class);
                     Bundle bundle = new Bundle();
-                    bundle.putString("request", new Gson().toJson(requestList.get(position)));
-                    //bundle.putInt( "position", position );
+                    //bundle.putString("request", new Gson().toJson(requestList.get(position)));
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java]: PLEASE DELETE THIS LINE

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java [hunk 11]
```diff
@@ -8,18 +8,18 @@
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.view.View;
+import android.widget.Button;
 import android.widget.ImageView;
 import android.widget.TextView;
 import android.widget.Toast;
 
+import com.google.gson.Gson;
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java:16]: dem imports ;)

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java [hunk 12]
```diff
@@ -62,7 +62,22 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // unpacking the bundle to get the position of request
         Bundle bundle = getIntent().getExtras();
-        request = new Gson().fromJson(bundle.getString("request"), Request.class);
+        //request = new Gson().fromJson(bundle.getString("request"), Request.class);
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java]: delete this now

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java [hunk 13]
```diff
@@ -319,6 +334,8 @@ public void onClick(DialogInterface dialog, int which) {
         } else {
             rc.addDriver(request, loggedInUser);
             Toast.makeText(this, "Made an offer.", Toast.LENGTH_SHORT).show();
+            view.setEnabled(false); // Make the button un-clickable after offering.
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/DriverViewRequestActivity.java]: great comments

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java [hunk 14]
```diff
@@ -17,18 +18,45 @@
...
+        if (requestsWhereOffered == null){
+            requestsWhereOffered = new RequestList();
         }
+        if (searchResult == null) {
+            searchResult = new RequestList();
+        }
+    }
+
+    /** Returns an instance of all requests where the user has offered to drive */
+    public RequestList getOffersInstance() {
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RequestController.java:48]: loving the comments

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/RiderRequestActivity.java [hunk 15]
```diff
@@ -61,10 +55,10 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // unpacking the bundle to get the position of request
         Bundle bundle = getIntent().getExtras();
-        //int position = bundle.getInt("position");
-        //request = rc.getResult().get(position);
+        int position = bundle.getInt("position");
+        request = rc.getRiderInstance().get(position);
 
-        request = new Gson().fromJson( bundle.getString("request"), Request.class );
+        //request = new Gson().fromJson( bundle.getString("request"), Request.class );
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/RiderRequestActivity.java]: delete this line?

File: app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchResultsActivity.java [hunk 16]
```diff
@@ -36,6 +36,7 @@ protected void onCreate(Bundle savedInstanceState) {
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                 Intent intent = new Intent(SearchResultsActivity.this, DriverViewRequestActivity.class);
                 Bundle bundle = new Bundle();
+                bundle.putInt( "position", -1 );
```
Comment (MandyMeindersma) [inline app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchResultsActivity.java:47]: ? what does this do? add a comment?

**Review by MandyMeindersma — CHANGES_REQUESTED** [review]
Once you look at all these comments could you see what was up with that weird crashing error. 

Other than that you did a wonerful amazing job and I am so proud of you

**Comment by Hreherch** [comment]
@MandyMeindersma I'm not going to fix the space because that's a simple refactor right click action and we can do that at the end and I don't want merge conflicts. But I think I'm done, test it out!

**Review by MandyMeindersma — APPROVED** [review]

**Pull request merged**